DFC.Digital/DFC.Digital.Repository.SitefinityCMS/Modules/CompositeUI/DynamicContentConverter.cs

```diff
@@ -0,0 +1,333 @@
+
+                // How To Become section
+                HowToBecomeData = htbContentPropertyConverter.ConvertFrom(content),
+                RelatedCareersData = GetRelatedCareersData(content, Constants.RelatedCareerProfiles),
+                Restrictions = GetRestrictions(content, RelatedRestrictionsField),
+                OtherRequirements = dynamicContentExtensions.GetFieldValue<Lstring>(content, OtherRequirementsField),
+                DynamicTitlePrefix = dynamicContentExtensions.GetFieldChoiceLabel(content, nameof(JobProfileMessage.DynamicTitlePrefix)),
+                DigitalSkillsLevel = dynamicContentExtensions.GetFieldChoiceLabel(content, nameof(JobProfileMessage.DigitalSkillsLevel)),
+            };
+
+            //What You will do section
+            jobProfileMessage.WhatYouWillDoData = GetWYDRelatedData(content);
+
+            //Related Skills Data
+            jobProfileMessage.RelatedSkills = GetSocMatrixSkills(content, Constants.RelatedSkills);
+
+            //Get SOC Code data
+            var socItem = dynamicContentExtensions.GetRelatedItems(content, Constants.SocField, 1).FirstOrDefault();
+
+            //SocCode Data
+            jobProfileMessage.SocCodeData = GenerateSocData(socItem);
+
+            //Working Pattern Details
+            jobProfileMessage.WorkingPatternDetails = MapClassificationData(content.GetValue<TrackedList<Guid>>(Constants.WorkingPatternDetail));
+
+            //Working Hours Details
+            jobProfileMessage.WorkingHoursDetails = MapClassificationData(content.GetValue<TrackedList<Guid>>(Constants.WorkingHoursDetail));
+
+            //Working Pattern
+            jobProfileMessage.WorkingPattern = MapClassificationData(content.GetValue<TrackedList<Guid>>(Constants.WorkingPattern));
+
+            //Hidden Alternative Title
+            jobProfileMessage.HiddenAlternativeTitle = MapClassificationData(content.GetValue<TrackedList<Guid>>(Constants.HiddenAlternativeTitle));
+
+            //Job Profile Specialism
+            jobProfileMessage.JobProfileSpecialism = MapClassificationData(content.GetValue<TrackedList<Guid>>(Constants.JobProfileSpecialism));
+
+            if (socItem != null)
+            {
+                jobProfileMessage.SocLevelTwo = dynamicContentExtensions.GetFieldValue<Lstring>(socItem, Constants.SOCCode).ToString().Substring(0, 2);
+            }
+
+            jobProfileMessage.LastModified = dynamicContentExtensions.GetFieldValue<DateTime>(content, nameof(JobProfileMessage.LastModified));
+            jobProfileMessage.CanonicalName = dynamicContentExtensions.GetFieldValue<Lstring>(content, nameof(JobProfileMessage.Title)).ToLower();
+            jobProfileMessage.JobProfileCategories = GetJobCategories(dynamicContentExtensions.GetFieldValue<IList<Guid>>(content, RelatedJobProfileCategoriesField));
+            return jobProfileMessage;
+        }
```

Comment:
Is this not difficult to debug? When there are failures it will be hard to know which of those props fail, better to split that in the next PR

Reply:
Will do it in the next PR